plugins/Goals/Controller.php

```diff
@@ -129,7 +130,7 @@ public function addNewGoal()
         $this->setGeneralVariablesView($view);
         $this->setGoalOptions($view);
         $view->onlyShowAddNewGoal = true;
-        $view->ecommerceEnabled = $this->site->isEcommerceEnabled();
+        $view->ecommerceEnabled = EcommerceEnabledSetting::getInstance($this->site->getId())->getValue() === 1;
```

Comment:
Controller has access to `idSite` which it uses to create the `$site` object, so we can rely on that.

```suggestion
        $view->ecommerceEnabled = EcommerceEnabledSetting::getInstance($this->idSite)->getValue() === 1;
```

Reply:
I've reverted this change, I've moved the ecommerce setting check to the sitesManager API for creating a site, so the check is done on load for a site. This still has an additional sql query to re-retrieve the ecommerce value, but I think it should be fine.